### PR TITLE
Use channels for uploads and increase chunk size

### PR DESCRIPTION
**Pull request by iskradelta**
For https://github.com/nextcloud/android/issues/348

Faster uploads, yay.

Though Im still disappointed, but 4-5megabytes/s is good enough, pushed it from 800-900kbytes/s previously... but could have been just due to testing this during different days.

### CODE REVIEW

File: src/com/owncloud/android/lib/common/network/ChunkFromFileChannelRequestEntity.java [hunk 1]
```diff
@@ -124,48 +129,42 @@ public void setmTransferred(long value) {
...
+                    if (size == 0) {
+                        size = -1;
                     }
+                    mDataTransferListener.onTransferProgress(transferredBytes, mTransferred, size, mFile.getAbsolutePath());
                 }
             }
 
-        } catch (IOException io) {
+        } catch (final IOException io) {
+            Log_OC.d(TAG, "Woopsie an io ", io);
```
Comment (AndyScherzinger) [inline src/com/owncloud/android/lib/common/network/ChunkFromFileChannelRequestEntity.java]: As for the logging, the if/elseif/else should have the respective log statements while the last else should have some generic message :wink:

**Comment by AndyScherzinger** [comment]
Awesome @iskradelta - will the chunk size then be file-size aware since smaller files like photos or any other should still use smaller chunk size due to the fact that some users might operate on slow networks with data plans so this will impact their cost depending on the amount of failed/retried chunks.

@tobiasKaminsky please have a look since you are the owner of the beta :smiley:

**Comment by mario** [comment]
@AndyScherzinger @tobiasKaminsky close this PR if it's no longer being worked on? :-/ Once we're out with 1.4.2, I'll write up what I want to do for 1.5 regarding this.

**Comment by AndyScherzinger** [comment]
In general fine with me. @iskradelta fancy some collab with @mario on this topic? :)

**Comment by iskradelta** [comment]
Whats the plan, what next?

**Comment by AndyScherzinger** [comment]
> Whats the plan, what next?

Any thought @mario? I won't have the time in the near future to support this issue coding wise, plus you already thought about a solution here :)

**Pull request closed without merging**